### PR TITLE
[vue] Add error handling toast

### LINKED ISSUE #16393: When there is a backend error, no error message shows up in the front end.
##### **Overview of the issue**

When there is a backend error, no error message shows up in the front end. The BadRequestAlertException is thrown in the backend. The following error messages show up on the browser console, 
```
createError.js:16 
Uncaught (in promise) Error: Request failed with status code 400
    at createError (createError.js:16)
    at settle (settle.js:17)
    at XMLHttpRequest.onloadend (xhr.js:54)

createError
@
createError.js:16

settle
@
settle.js:17

onloadend
@
xhr.js:54

Promise.then (async)



push../node_modules/ts-loader/index.js??clonedRuleSet-1[0].rules[0].use[0]!./src/main/webapp/app/entities/job-candidate-repo/job-candidate-repo-update.component.ts?vue&type=script&lang=ts&.JobCandidateRepoUpdate.save
@
job-candidate-repo-u…te.component.ts:106

submit
@
job-candidate-repo-update.vue?cd42:14

invokeWithErrorHandling
@
vue.esm.js:1872

invoker
@
vue.esm.js:2197

original._wrapper
```
But, there isn't an error message in the front end. 

I have reported the error before due to another error type 500. The fix likely doesn't address the root problem. I still observe the problem after upgrading to 7.2.0.

##### **Motivation for or Use Case**

User experience

##### **Related issues**

#17522

##### **Suggest a Fix**

The BadRequestAlertException perhap needs to be handled.

##### **JHipster Version(s)**

7.1.0, 7.2.0

##### **JHipster configuration**
```

{
  "generator-jhipster": {
    "applicationType": "monolith",
    "baseName": "myapp",
    "jhipsterVersion": "7.1.0",
    "skipClient": false,
    "skipServer": false,
    "skipUserManagement": true,
    "skipCheckLengthOfIdentifier": false,
    "skipFakeData": false,
    "jhiPrefix": "jhi",
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "testFrameworks": ["cypress", "gatling"],
    "blueprints": [],
    "otherModules": [],
    "pages": [],
    "creationTimestamp": 1626276896721,
    "serviceDiscoveryType": false,
    "reactive": false,
    "authenticationType": "oauth2",
    "packageName": "com.mycompany.myteams.myapp",
    "serverPort": "8080",
    "cacheProvider": "ehcache",
    "enableHibernateCache": true,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "postgresql",
    "buildTool": "maven",
    "serverSideOptions": [],
    "websocket": false,
    "searchEngine": false,
    "messageBroker": false,
    "enableSwaggerCodegen": false,
    "clientFramework": "vue",
    "withAdminUi": true,
    "clientTheme": "cosmo",
    "clientThemeVariant": "primary",
    "enableTranslation": true,
    "nativeLanguage": "en",
    "packageFolder": "com/mycompany/myteams/myapp",
    "devServerPort": 9060,
    "clientPackageManager": "npm",
    "languages": ["en", "de"],
    "entities": ["JobCandidateRepo", "RepositoryTemplate"],
    "lastLiquibaseTimestamp": 1629745008000
  }
}
```

##### **Browsers and Operating System**

- [ ] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by swarajsaaj**
Add error handling for multiple scenarios in Vue with a toast.

Approach:- The errors are being shown in the toast through a service instead of interceptor to allow users more choice over generated code to pick and choose errors they wish to show/handle.

Fix #16393

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/client/templates/vue/src/main/webapp/app/shared/alert/alert.service.ts.ejs [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import Vue from 'vue';
+
+export default class AlertService {
```
Comment (pascalgrimaud) [inline generators/client/templates/vue/src/main/webapp/app/shared/alert/alert.service.ts.ejs:3]: is it possible to have alert.service.spec.ts too plz ?

Comment (swarajsaaj) [reply generators/client/templates/vue/src/main/webapp/app/shared/alert/alert.service.ts.ejs:3]: added for translation/entity-header scenarios in 02c6214  :)

**Comment by swarajsaaj** [comment]
Thanks for review and merge @pascalgrimaud

**Comment by swarajsaaj** [comment]
bounty claimed at https://opencollective.com/generator-jhipster/expenses/51929 :)

**Comment by pascalgrimaud** [comment]
@swarajsaaj : approved !

**Pull request merged**